dcs/country.py

```diff
@@ -81,6 +81,20 @@ def find_group(self, group_name, search="exact"):
                     return group
         return None
 
+    def find_group_by_id(self, group_id: int) -> Optional[Group]:
+        groups = [self.vehicle_group,
+                  self.ship_group,
+                  self.plane_group,
+                  self.helicopter_group,
+                  self.static_group]
+        for search_group in groups:
+            assert isinstance(search_group, list)
```

Comment:
This looks unnecessary, and would prevent the type checker from doing its job if the types ever changed.

Reply:
@DanAlbert  - mypy is not happy if it's not there. You wanna argue with it? :-)

![image](https://github.com/pydcs/dcs/assets/152709000/c8948075-3572-44ba-86e8-ef9892a349d4)

Reply:
I kinda do. I'll pull your branch and have a look, because it doesn't make sense to me. Wouldn't be the first time there's a bug in mypy but I don't mind looking.

Reply:
I think it has to do with mypy's type inference. If I explicitly type `groups` as `list[list[Group]]` I get:

```
dcs\country.py:86: error: List item 0 has incompatible type "List[VehicleGroup]"; expected "Group[Any, Any]"  [list-item]
dcs\country.py:87: error: List item 1 has incompatible type "List[ShipGroup]"; expected "Group[Any, Any]"  [list-item]
dcs\country.py:88: error: List item 2 has incompatible type "List[PlaneGroup]"; expected "Group[Any, Any]"  [list-item]
dcs\country.py:89: error: List item 3 has incompatible type "List[HelicopterGroup]"; expected "Group[Any, Any]"  [list-item]
dcs\country.py:90: error: List item 4 has incompatible type "List[StaticGroup]"; expected "Group[Any, Any]"  [list-item]
dcs\country.py:92: error: "Group[Any, Any]" has no attribute "__iter__"; maybe "__str__"? (not iterable)  [attr-defined]
```

If I type it as `list[Sequence[Group]]` it works fine, and unlike the assert, the type checker will still prove it correct and emit an error if the type ever changes. That's `from typing import Sequence`, btw :) (or `from collections.abc`... but I think we still support python versions old enough that we need to use the deprecated type from `typing`?)

Reply:
@DanAlbert  - Resolved by adding a type hint to groups.